web/src/stream_edit.ts

```diff
@@ -747,6 +749,42 @@ export function initialize(): void {
         $(".dialog_submit_button").attr("data-stream-id", stream_id);
     });
 
+    $("#channels_overlay_container").on("click", ".reactivate", function (this: HTMLElement, e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const stream_id = get_stream_id(this);
+        function do_unarchive_stream(): void {
+            channel.patch({
+                url: `/json/streams/${stream_id}`,
+                data: {is_archived: false},
```

Comment:
This looks like it's missing a loading spinner; the similar groups code seems to have included that correctly.